File: src/main/scala/li/cil/oc/common/inventory/MultiTank.scala
package li.cil.oc.common.inventory

import net.minecraftforge.fluids.IFluidTank

// TODO Move to api.internal in 1.5, make internal.Robot extend this.
trait MultiTank {
  def tankCount: Int

  def getFluidTank(index: Int): IFluidTank
}

File: src/main/java/li/cil/oc/api/internal/MultiTank.java
package li.cil.oc.api.internal;

import net.minecraftforge.fluids.IFluidTank;

/**
 * Implemented by objects with multiple internal tanks.
 * <p/>
 * This is specifically for containers where the side does not matter when
 * accessing the internal tanks, only the index of the tank; unlike with the
 * {@link net.minecraftforge.fluids.IFluidHandler} interface.
 */
public interface MultiTank {
    /**
     * The number of tanks currently installed.
     */
    int tankCount();

    /**
     * Get the installed fluid tank with the specified index.
     *
     * @param index the index of the tank to get.
     * @return the tank with the specified index.
     */
    IFluidTank getFluidTank(int index);
}
